NuRadioReco/examples/RNO_energy_reconstruction/T03_electric_field_reco.py

```diff
@@ -0,0 +1,123 @@
+args = parser.parse_args()
+
+noise_level = args.noise_level * units.mV
+sampling_rate = 5. * units.GHz
+vertex_reco_passband = [.1, .3]
+efield_reco_passband = [.13, .5]
+ice = NuRadioMC.utilities.medium.get_ice_model('greenland_simple')
+event_reader = NuRadioReco.modules.io.NuRadioRecoio.NuRadioRecoio([args.input_file])
+channel_resampler = NuRadioReco.modules.channelResampler.channelResampler()
+efield_resampler = NuRadioReco.modules.electricFieldResampler.electricFieldResampler()
+event_writer = NuRadioReco.modules.io.eventWriter.eventWriter()
+event_writer.begin(args.output_file)
+det = NuRadioReco.detector.generic_detector.GenericDetector(
+    json_filename=args.detector_file,
+    antenna_by_depth=False
+)
+channel_bandpass_filter = NuRadioReco.modules.channelBandPassFilter.channelBandPassFilter()
+efield_bandpass_filter = NuRadioReco.modules.electricFieldBandPassFilter.electricFieldBandPassFilter()
+channel_pulse_finder = NuRadioReco.modules.channelPulseFinderSimulator.channelPulseFinderSimulator()
+channel_pulse_finder.begin(noise_level=noise_level)
+"""
+We create an electric field template to be used to find the radio pulse. Pretty much any
+short pulse will do, so we create a delta pulse by setting a constant spectrum, apply a filter, and shift the pulse to
+be in the middle of the trace.
+"""
+spec = np.ones(int(128 * sampling_rate + 1)) * bandpass_filter.get_filter_response(np.fft.rfftfreq(int(256 * sampling_rate), 1. / sampling_rate),    [.1, .3], 'butter', 10)
+efield_template = NuRadioReco.framework.base_trace.BaseTrace()
+efield_template.set_frequency_spectrum(spec, sampling_rate)
+efield_template.apply_time_shift(20. * units.ns, True)
+
+if not os.isdir('plots/efield_reco'):
+    os.mkdir('plots/efield_reco')
+ift_efield_reconstructor = NuRadioReco.modules.iftElectricFieldReconstructor.iftElectricFieldReconstructor.IftElectricFieldReconstructor()
+ift_efield_reconstructor.begin(
+    electric_field_template=efield_template,
+
+    passband=efield_reco_passband,
+    n_samples=10,
+    n_iterations=5,
+    phase_slope='negative',
+    energy_fluence_passbands=[
+        [.13, .2],
+        [.13, .25],
+        [.13, .3]
+    ],
+    slope_passbands=[
+        [[.13, .2], [.2, .3]],
+        [[.13, .25], [.25, .5]],
+        [[.13, .3], [.3, .5]],
+    ],
+    debug=True,
+    plot_folder='plots/efield_reco'
+)
+time_offset_calculator = NuRadioReco.modules.channelTimeOffsetCalculator.channelTimeOffsetCalculator()
+time_offset_calculator.begin(
+    electric_field_template=efield_template,
+    medium=ice
+)
+channel_props_from_neighbor = NuRadioReco.modules.channelSignalPropertiesFromNeighbors.channelSignalPropertiesFromNeighbors()
+for i_event, event in enumerate(event_reader.get_events()):
+    print('Event {}, Run={}, ID={}'.format(i_event, event.get_run_number(), event.get_id()))
+    station = event.get_station(11)
```

Comment:
I think it should be clear that 11 is the station ID. Just using the first station in the detector description is definitely a bad idea, because there might be other stations in there. The meaning of the channel groups is explained in the docstring of the module that uses it. I think this is fine as it is.